### PR TITLE
Pr notapatch refactor recommendation last acs

### PR DESCRIPTION
**Pull request by notapatch**
## Description

[This Pr refactors the code before fixes the last AC in the trello card.](https://trello.com/c/F5YRGSA0/1219-reviewer-is-able-to-view-comments-against-legislation-by-officer)
There's also missing fucntionality of next/prev added with component.

[PlanningApplication.recommendation (from recommendations.last) - refactor](https://github.com/unboxed/bops/commit/4147aa691689910d5405b0425635d2e6a9551a8f)
- recommendation from recommendations.last
  - I just wanted to hide that there was an array and that they wanted the last and instead used the method that simply returned the last. 

[Planning application interface for #last_recommendation_accepted? - refactor](https://github.com/unboxed/bops/commit/6c31b02c6a370757b0ea2a78a91a9a4ed12d0ca8)
- My code to access if recommendation accepted was using 3 dots - in short the structure was over exposed. The other effect of the exposure made it hard to test. So I added a method #last_recommendation_accepted? to planning application and made the code for the other reviewing code use it.

[Review policy class navigation component - missing functinality](https://github.com/unboxed/bops/commit/e01563b4846ff4a9c2e8871e71ca7a2b14d88782)

- component for Next previous navigation between policy classes

[ReviewPolicyClass errors if you save after accepting recommendation - last AC](https://github.com/unboxed/bops/commit/6141de4234471cdb1c8b928934d8e6e10df90fc1)
- used the method #last_recommendation_accepted? to solve the last AC

### CODE REVIEW

File: app/models/planning_application.rb [hunk 1]
```diff
@@ -486,6 +486,12 @@ def recommendation
     @recommendation ||= recommendations.last
   end
 
+  def last_recommendation_accepted?
```
Comment (allbecauseyoutoldmeso) [inline app/models/planning_application.rb:489]: I think we can use this [here](https://github.com/unboxed/bops/blob/main/app/models/assessment_details_review.rb#L105) as well 🎉

**Review by allbecauseyoutoldmeso — APPROVED** [review]

File: app/components/review_policy_classes/navigation_component.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+module ReviewPolicyClasses
+  class NavigationComponent < ViewComponent::Base
+    def initialize(policy_class:)
+      @policy_class = policy_class
+    end
+
+    def multiple_policy_classes?
+      @policy_class.planning_application.policy_classes.count > 1
+    end
+  end
```
[the remark this answers is not in the document]
Comment (notapatch) [reply app/components/review_policy_classes/navigation_component.rb:12]: I think of the component as an object and I don't want to give readers into the object that doesn't need to be. How do you see it?

File: app/models/planning_application.rb (hunk 1 above)
Comment (notapatch) [reply app/models/planning_application.rb:489]: I think it's covered in a later commit but I will double check.....

File: app/components/review_policy_classes/navigation_component.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (allbecauseyoutoldmeso) [reply app/components/review_policy_classes/navigation_component.rb:14]: Ah, so I was thinking of a private `attr_reader` that wraps `@policy_class`, and then you use that within the class. Just as a way of avoiding having instance variables floating around, with the potential to be accidentally re-assigned.

But I think this is a habit I was inculcated with way back when at coding bootcamp, so maybe ignore me 🙈

[the remark this answers is not in the document]
Comment (allbecauseyoutoldmeso) [reply app/components/review_policy_classes/navigation_component.rb:14]: One thing I find a bit weird about components, although I suppose it makes sense, is that the `html` file has access to the private methods in the ruby class.

[the remark this answers is not in the document]
Comment (notapatch) [reply app/components/review_policy_classes/navigation_component.rb:14]: **[Implementation](https://viewcomponent.org/guide/getting-started.html)**
```
A ViewComponent is a Ruby file and corresponding template file with the same base name:

# app/components/example_component.rb
class ExampleComponent < ViewComponent::Base
  def initialize(title:)
    @title = title
  end
end
<%# app/components/example_component.html.erb %>
<span title="<%= @title %>"><%= content %></span>
```

Reading through the example code from the site. Some have attr_readers and some don't. Here's an example where they haven't. 🤔

[the remark this answers is not in the document]
Comment (allbecauseyoutoldmeso) [reply app/components/review_policy_classes/navigation_component.rb:14]: Let's go with either approach is 100% valid 😁

[the remark this answers is not in the document]
Comment (notapatch) [reply app/components/review_policy_classes/navigation_component.rb:14]: > One thing I find a bit weird about components, although I suppose it makes sense, is that the html file has access to the private methods in the ruby class.

Yes, I see the ruby and the associated html as one thing. Better still with a utility css like Tailwind where the css you need would go into the component html file. At that point everything you need to know about the component is in two files and the component becomes an element of reuse.

**Pull request merged**